tools/metric_frontend_bound

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright (C) 2018 Intel Corporation
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files
+# (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge,
+# publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so, 
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+# OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+# OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# SPDX-License-Identifier: MIT
+
+SCRIPTS_DIR=`dirname $0`
+source ${SCRIPTS_DIR}/utils.sh
+
+function init_frontend_bound() {
+  local local_pmu_array=(cpu_clk_unhalted.thread_any idq_uops_not_delivered.core)
+  local local_pmus
+  for item in ${local_pmu_array[*]}
+  do
+    if [ "x${local_pmus}" == "x" ]; then
+      local_pmus="$item"
+    else
+      local_pmus="$local_pmus,$item"
+    fi
+  done
+  echo $local_pmus
+}
+
+function calc_frontend_bound() {
+  local perf_data_file="$1"
+  local metric_name="metric_frontend_bound>"
+  echo
+  echo "================================================="
+  echo "Final ${metric_name}"
+  echo "--------------------------------------------------"
+  echo "FORMULA: ${metric_name} = 100*b/(4*(a/threads))"
+  echo "         where, a=cpu_clk_unhalted.thread_any"
+  echo "                b=idq_uops_not_delivered.core"
+  echo "================================================="
+
+  local a=`return_pmu_value "cpu_clk_unhalted.thread_any" $perf_data_file `
```

Comment:
Extra space after $perf_data_file

Reply:
> Thanks @uttampawar for the PR.
> 
> Here are some comments:
> 
> * What if HT is not ON? Will threads be correct? You may want to get this programatically.
> * Formulas are available on https://download.01.org/perfmon/TMA_Metrics.xlsx
> 
> ```
> FE BOUND = IDQ_UOPS_NOT_DELIVERED.CORE / SLOTS
> SLOTS = #Pipeline_Width  * CORE_CLKS
> CORE_CLKS = CPU_CLK_UNHALTED.THREAD
> ```
> 
> * In future we want to add some checks about the architecture. Some other formulas work only on Skylake and beyond. This one is fine.
> 
> --

 I realize it yesterday while working on this script. This needs to be addressed for sure one for getting up to date architecture information, exact counter names and formulas to getting various metrics.